### PR TITLE
HTS-1133 Added trimmed email to validate and to AccountHolderControll…

### PR DESCRIPTION
**Pull request by Jackie1988**
…er onSubmit

### CODE REVIEW

File: test/uk/gov/hmrc/helptosavefrontend/forms/EmailValidationSpec.scala [hunk 1]
```diff
@@ -16,18 +16,21 @@
 
 package uk.gov.hmrc.helptosavefrontend.forms
 
+import cats.data.NonEmptyList
+import cats.data.Validated.Valid
```
Comment (sureshhmrc) [inline test/uk/gov/hmrc/helptosavefrontend/forms/EmailValidationSpec.scala]: are we using these imports ?

Comment (Jackie1988) [reply test/uk/gov/hmrc/helptosavefrontend/forms/EmailValidationSpec.scala]: Valid is being used, the other isn't so I have removed it

File: app/uk/gov/hmrc/helptosavefrontend/controllers/AccountHolderController.scala [hunk 2]
```diff
@@ -76,19 +76,23 @@ class AccountHolderController @Inject() (val helpToSaveService:          HelpToS
...
+                        name,
+                        SeeOther(routes.AccountHolderController.getCheckYourEmail().url),
+                        params ⇒ routes.AccountHolderController.emailVerifiedCallback(params.encode()).url,
+                        _ ⇒ SeeOther(routes.EmailVerificationErrorController.verifyEmailErrorTryLater().url),
+                        isNewApplicant = false)
+                    ).leftMap { e ⇒
+                      logger.warn(s"Could not write pending email to session cache: $e")
+                      SeeOther(routes.EmailVerificationErrorController.verifyEmailErrorTryLater().url)
+                    }.merge
+                case Left(_) ⇒ SeeOther(routes.AccountHolderController.getUpdateYourEmailAddress().url)
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/AccountHolderController.scala]: adding logging helps debugging ?

Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosavefrontend/controllers/AccountHolderController.scala]: ok yeah sure

**Pull request merged**